src/main/java/no/nav/foreldrepenger/oversikt/innhenting/inntektsmelding/InntektsmeldingTjeneste.java

```diff
@@ -23,13 +26,32 @@ public InntektsmeldingTjeneste(InntektsmeldingerRepository inntektsmeldingerRepo
         // CDI
     }
 
-    List<InntektsmeldingDto> inntektsmeldinger(Saksnummer saksnummer) {
-        return inntektsmeldingerRepository.hentFor(Set.of(saksnummer)).stream()
-                .map(InntektsmeldingTjeneste::tilInntektsmeldingDto)
-                .toList();
+    List<FpOversiktInntektsmeldingDto> inntektsmeldinger(Saksnummer saksnummer) {
+        return inntektsmeldingerRepository.hentFor(Set.of(saksnummer)).stream().map(InntektsmeldingTjeneste::tilInntektsmeldingDto).toList();
     }
 
-    private static InntektsmeldingDto tilInntektsmeldingDto(Inntektsmelding inntektsmelding) {
-        return new InntektsmeldingDto(inntektsmelding.mottattTidspunkt() == null ? inntektsmelding.innsendingstidspunkt() : inntektsmelding.mottattTidspunkt());
+    private static FpOversiktInntektsmeldingDto tilInntektsmeldingDto(Inntektsmelding inntektsmelding) {
+        if (inntektsmelding instanceof InntektsmeldingV2 inntektsmeldingV2) {
+            var naturalytelser = inntektsmeldingV2.bortfalteNaturalytelser()
+                .stream()
+                .map(n -> new FpOversiktInntektsmeldingDto.NaturalYtelse(n.fomDato(), n.tomDato(), n.beloepPerMnd(), n.type()))
+                .toList();
+            var refusjon = inntektsmeldingV2.refusjonsperioder()
+                .stream()
+                .map(r -> new FpOversiktInntektsmeldingDto.Refusjon(r.fomDato(), r.refusjonsbeløpMnd()))
+                .toList();
+            return new FpOversiktInntektsmeldingDto(2, inntektsmeldingV2.erAktiv(), inntektsmeldingV2.stillingsprosent(), inntektsmeldingV2.inntektPrMnd(), inntektsmeldingV2.refusjonPrMnd(),
+                inntektsmeldingV2.arbeidsgiverNavn(), inntektsmeldingV2.journalpostId(), inntektsmeldingV2.kontaktpersonNavn(),
+                inntektsmeldingV2.kontaktpersonNummer(), inntektsmeldingV2.mottattTidspunkt(),
+                inntektsmeldingV2.startDatoPermisjon(), naturalytelser, refusjon);
+        }
+        if (inntektsmelding instanceof InntektsmeldingV1 inntektsmeldingV1) {
```

Comment:
Denne kunne nesten fjernes egentlig. I frontend bryr jeg meg ikke om v1, så den ignoreres. Men kan vel leve til vi har byttet ut alle v1 med v2 i mellomlagringen?

Reply:
Da vil vel alle inntektsmeldingsinnslag på tidslinja bli borte i mellomtiden?

Reply:
Slik jeg har tenkt det nå så vil v1 fremdeles serveres til søknad-api slik at de brukes i tidslinja. Ny controller i api vil også hente alle IM (v1+v2) også gjør frontend en filter på type slik at det kun er v2 som brukes til visning. Mens v1 fremdeles vil brukes av søknad-api for å generere tidslinje